gammapy/irf/background.py

```diff
@@ -132,4 +133,140 @@ def to_table(self):
+
+    def __init__(self, energy_lo, energy_hi,
+                 offset_lo, offset_hi,
+                 data, meta=None, interp_kwargs=None):
+
+        if interp_kwargs is None:
+            interp_kwargs = self.default_interp_kwargs
+        axes = [
+            BinnedDataAxis(
+                energy_lo, energy_hi,
+                interpolation_mode='log', name='energy'),
+            BinnedDataAxis(
+                offset_lo, offset_hi,
+                interpolation_mode='linear', name='offset'),
+        ]
+        self.data = NDDataArray(axes=axes, data=data,
+                                interp_kwargs=interp_kwargs)
+        self.meta = OrderedDict(meta) if meta else OrderedDict()
+
+    def __str__(self):
+        ss = self.__class__.__name__
+        ss += '\n{}'.format(self.data)
+        return ss
+
+    @classmethod
+    def from_table(cls, table):
+        """Read from `~astropy.table.Table`."""
+        # Spec says key should be "BKG", but there are files around
+        # (e.g. CTA 1DC) that use "BGD". For now we support both
+        if 'BKG' in table.colnames:
+            bkg_name = 'BKG'
+        elif 'BGD' in table.colnames:
+            bkg_name = 'BGD'
+        else:
+            raise ValueError('Invalid column names. Need "BKG" or "BGD".')
+
+        # Currently some files (e.g. CTA 1DC) contain unit in the FITS file
+        # '1/s/MeV/sr', which is invalid ( try: astropy.unit.Unit('1/s/MeV/sr')
+        # This should be corrected.
+        # For now, we hard-code the unit here:
+        data_unit = u.Unit('s-1 MeV-1 sr-1')
+        return cls(
+            energy_lo=table['ENERG_LO'].quantity[0],
+            energy_hi=table['ENERG_HI'].quantity[0],
+            offset_lo=table['THETA_LO'].quantity[0],
+            offset_hi=table['THETA_HI'].quantity[0],
+            data=table[bkg_name].data[0] * data_unit,
+            meta=table.meta,
+        )
+
+    @classmethod
+    def from_hdulist(cls, hdulist, hdu='BACKGROUND'):
+        """Create from `~astropy.io.fits.HDUList`."""
+        fits_table = hdulist[hdu]
+        table = fits_table_to_table(fits_table)
+        return cls.from_table(table)
+
+    @classmethod
+    def read(cls, filename, hdu='BACKGROUND'):
+        """Read from file."""
+        filename = make_path(filename)
+        hdulist = fits.open(str(filename))
+        return cls.from_hdulist(hdulist, hdu=hdu)
+
+    def to_table(self):
+        """Convert to `~astropy.table.Table`."""
+        meta = self.meta.copy()
+        table = Table(meta=meta)
+
+        table['THETA_LO'] = self.data.axis('offset').lo[np.newaxis]
+        table['THETA_HI'] = self.data.axis('offset').hi[np.newaxis]
+        table['ENERG_LO'] = self.data.axis('energy').lo[np.newaxis]
+        table['ENERG_HI'] = self.data.axis('energy').hi[np.newaxis]
+        table['BKG'] = self.data.data[np.newaxis]
+        return table
+
+    def to_hdulist(self, name='BACKGROUND'):
```

Comment:
Given that our IRFs serialise into a single HDU, having a method that returns a HDU list isn't very useful, no?
Also it's bad to have different methods for Background2D / Background3D.

This boilerplate code will probably anyways go to a base class (see #1157), but for now, I'd suggest to  make Background2D and Background3D the same.